app/models/piece.rb

```diff
@@ -2,8 +2,58 @@ class Piece < ActiveRecord::Base
   belongs_to :users
   belongs_to :games
 
-def is_obstructed?
-	return false
-end
+	def is_obstructed?(x, y)
+
+		#check moving direction
+		path = moving_direction(x, y)
+
+		#path is horizontal from left to right
+		if path == 'horizontal' && x_cord < x
+			(x_cord + 1).upto(x - 1) do |changing_x|
```

Comment:
@AlekstheRealist that is why I included "x_cord <x" on line 11. The other way around should be "x_cord>x"